AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/InitialSync/SyncEventEmitterTests.swift

```diff
@@ -0,0 +1,188 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSPluginsCore
+@testable import AWSDataStoreCategoryPlugin
+
+class SyncEventEmitterTests: XCTestCase {
+    var initialSyncOrchestrator: MockAWSInitialSyncOrchestrator?
+    var reconciliationQueue: MockAWSIncomingEventReconciliationQueue?
+    var syncEventEmitter: SyncEventEmitter?
+
+    override func setUp() {
+        super.setUp()
+
+        ModelRegistry.reset()
+        MockModelReconciliationQueue.reset()
+        MockAWSInitialSyncOrchestrator.reset()
+    }
+
+    /// - Given: A SyncEventEmitter
+    /// - When:
+    ///    - One model is registered
+    ///    - Perform an initial sync
+    /// - Then:
+    ///    - One modelSynced event should be received
+    ///    - One syncQueriesReady event should be received
+    func testModelSyncedAndSyncQueriesReadyWithOneModelRegistered() throws {
+        let modelSyncedReceived = expectation(description: "modelSynced received")
+        let syncQueriesReadyReceived = expectation(description: "syncQueriesReady received")
+
+        ModelRegistry.register(modelType: Post.self)
+        let testPost = Post(id: "1", title: "post1", content: "content", createdAt: .now())
+        let anyPost = AnyModel(testPost)
+        let anyPostMetadata = MutationSyncMetadata(id: "1",
+                                               deleted: false,
+                                               lastChangedAt: Int(Date().timeIntervalSince1970),
+                                               version: 1)
+        let anyPostMutationSync = MutationSync<AnyModel>(model: anyPost, syncMetadata: anyPostMetadata)
+        let postMutationEvent = try MutationEvent(untypedModel: testPost, mutationType: .create)
+
+        let listener = Amplify.Hub.publisher(for: .dataStore).sink { payload in
+            switch payload.eventName {
+            case HubPayload.EventName.DataStore.modelSynced:
+                guard let modelSyncedEventPayload = payload.data as? ModelSyncedEvent else {
+                    XCTFail("Couldn't cast payload data as ModelSyncedEvent")
+                    return
+                }
+                XCTAssertEqual(modelSyncedEventPayload.modelName, "Post")
+                XCTAssertTrue(modelSyncedEventPayload.isFullSync)
+                XCTAssertFalse(modelSyncedEventPayload.isDeltaSync)
+                XCTAssertEqual(modelSyncedEventPayload.added, 1)
+                XCTAssertEqual(modelSyncedEventPayload.updated, 0)
+                XCTAssertEqual(modelSyncedEventPayload.deleted, 0)
```

Comment:
can you build an `expectedMoySyncEventPayload` and then assert that the two objects are equal?

Reply:
In this case, I should make `ModelSyncedEvent` conform to Equatable. I guess that is not a problem?